test/test_distributions.py

```diff
@@ -522,6 +530,63 @@ def test_gamma_sample_grad(self):
                                        'rel error {}'.format(rel_error),
                                        'max error {}'.format(rel_error.max())]))
 
+    # This is a randomized test.
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_chi2_shape(self):
+        df = Variable(torch.exp(torch.randn(2, 3)), requires_grad=True)
+        df_1d = Variable(torch.exp(torch.randn(1)), requires_grad=True)
+        self.assertEqual(Chi2(df).sample().size(), (2, 3))
+        self.assertEqual(Chi2(df).sample_n(5).size(), (5, 2, 3))
+        self.assertEqual(Chi2(df_1d).sample_n(1).size(), (1, 1))
+        self.assertEqual(Chi2(df_1d).sample().size(), (1,))
+        self.assertEqual(Chi2(0.5).sample().size(), (1,))
+        self.assertEqual(Chi2(0.5).sample_n(1).size(), (1,))
+
+        def ref_log_prob(idx, x, log_prob):
+            d = df.data.view(-1)[idx]
+            expected = scipy.stats.chi2.logpdf(x, d)
+            self.assertAlmostEqual(log_prob, expected, places=3)
+
+        self._check_log_prob(Chi2(df), ref_log_prob)
+
+    # This is a randomized test.
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_chi2_sample(self):
+        set_rng_seed(0)
```

Comment:
This needs a reference to the note about testing that has been added in one of the recent PRs. @fritzo Does it also apply to the test below?

Reply:
Yes. @alicanb In #4357 I've updated all comments as suggested by Edward:
```diff
- # This is a randomized test.
  @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
  def test_chi2_sample(self):
-    set_rng_seed(0)
+    set_rng_seed(0)  # see Note [Randomized statistical tests]
```